plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -919,6 +931,26 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 		self->write_recovery_bank = address == self->fw_bank_addr[1];
 	}
 
+	/* has public key */
+	if (fu_device_has_private_flag(device, FU_GENESYS_USBHUB_FLAG_HAS_PUBLIC_KEY)) {
+		g_autofree gchar *guid = NULL;
+		if (!fu_memcpy_safe(self->public_key,
+				    sizeof(self->public_key),
+				    0, /* dst */
+				    buf,
+				    bufsz,
+				    self->fw_data_total_count, /* src */
+				    sizeof(self->public_key),
+				    error))
+			return FALSE;
+		guid = fwupd_guid_hash_data(self->public_key,
+					    sizeof(self->public_key),
+					    FWUPD_GUID_FLAG_NONE);
+		fu_device_add_instance_strup(device, "PUBKEY", guid);
+		if (!fu_device_build_instance_id(device, NULL, "USB", "VID", "PID", "PUBKEY", NULL))
+			return FALSE;
```

Comment:
you can do this alongside the other calls to fu_device_build_instance_id --  if `PUBKEY` isn't defined then it's a No-Op.

Reply:
Well, there is no other call `fu_device_build_instance_id()` (yet :p); I have the idea.

However, I got a segmentation fault if the instance is not defined.

I made the change below:

```
--- a/plugins/genesys/fu-genesys-usbhub-device.c
+++ b/plugins/genesys/fu-genesys-usbhub-device.c
@@ -949,10 +949,11 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
                                            sizeof(self->public_key),
                                            FWUPD_GUID_FLAG_NONE);
                fu_device_add_instance_strup(device, "PUBKEY", guid);
-               if (!fu_device_build_instance_id(device, NULL, "USB", "VID", "PID", "PUBKEY", NULL))
-                       return FALSE;
        }

+       if (!fu_device_build_instance_id(device, NULL, "USB", "VID", "PID", "PUBKEY", NULL))
+               return FALSE;
+
        /* have MStar scaler */
        if (fu_device_has_private_flag(device, FU_GENESYS_USBHUB_FLAG_HAS_MSTAR_SCALER))
                if (!fu_genesys_usbhub_device_mstar_scaler_setup(self, error))
```

Do you still want that change? and/or I fix the segmentation or ???

Reply:
done